pkg/controller/kieapp/defaults/defaults.go

```diff
@@ -127,26 +131,32 @@ func getEnvTemplate(cr *v1.KieApp) v1.EnvTemplate {
 	return envTemplate
 }
 
-func loadYaml(filename string, t v1.EnvTemplate) ([]byte, error) {
-	box := packr.NewBox("../../../../config")
-
-	if box.Has(filename) {
-		// important to parse template first, before unmarshalling into object
-		bytes, err := box.Find(filename)
-		if err != nil {
-			logrus.Error(err)
-			return nil, err
+func loadYaml(client client.Client, filename, namespace string, t v1.EnvTemplate) ([]byte, error) {
+	cmName, file := convertToConfigMapName(filename)
+	logrus.Debugf("Loading contents from %s in ConfigMap '%s'", file, cmName)
+	configMap := &corev1.ConfigMap{}
+	err := client.Get(context.TODO(), types.NamespacedName{Name: cmName, Namespace: namespace}, configMap)
+	if err != nil {
```

Comment:
Is there a way to distinguish between "I'm running a unit test or don't want configmaps" and "your attempt to create/update a configmap was unsuccessful and there is an error".

Too often, our software ignores errors, leading to poor usability. I'm thinking something went wrong and the customer is scratching their head, wondering why their configmap change isn't working as intended, not realizing it's being ignored in favor of the default one.

Reply:
the create configmap logic resides in the kieapp controller code. it will throw an error if there was an issue reconciling the configmap in question.
https://github.com/kiegroup/kie-cloud-operator/blob/3d7d81ecbc36df26e5b4efd42294bbe47db3750b/pkg/controller/kieapp/kieapp_controller.go#L558-L576

Reply:
Not sure I follow... I don't see where the error is thrown, but also back to line of code that I commented on... what are (all) the possible scenarios where err is not nil?

Reply:
its thrown in the `reconciler.createObj` function.

Reply:
hmm... so would you like something that specifies whether the CM doesn't exist vs an error on retrieval of said CM?

Reply:
this has been improved as you asked... if a configmap is unavailable, the KieApp deployment will fail and the user will see this in the logs -
`ERRO[0022] tommy5/kieconfigs-envs ConfigMap inaccessible, 'production' KieApp not deployed. Retrying...  `